internal/services/network/virtual_network_gateway_connection_resource.go

```diff
@@ -411,6 +412,11 @@ func resourceVirtualNetworkGatewayConnectionCreateUpdate(d *pluginsdk.ResourceDa
 		if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
 			return fmt.Errorf("waiting for updating Shared Key for %s: %+v", id, err)
 		}
+
+		// this LRO operation returns as 200 cause no wait actually, add a time.Sleep()
+		// see https://github.com/Azure/azure-rest-api-specs/issues/24232
+		// todo: remove time.Sleep once the issue resolved
+		time.Sleep(time.Second * 10)
```

Comment:
we should never rely on a sleep. please do a statewait instead to poll for when the create/update is actually finished